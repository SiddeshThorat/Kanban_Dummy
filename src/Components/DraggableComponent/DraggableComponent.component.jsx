import React,{useState} from 'react';
import './DraggableComponent.styles.css';
import { URL } from '../../Constants/url';

const DraggableComponent = ({task,onDragStart,editTask}) => {
    const [editStatus, changeEditStatus] = useState(false)
    const [changedTaskTitle, setTaskTitle] = useState(task.title);

    const enableEdit = () => {
        changeEditStatus(true)
    }

    const changeTaskTitle = (event) => {
        event.preventDefault();
        setTaskTitle(event.target.value)
    }

    const saveChanges = async () => {
        const options = {
            method: 'PUT',
            headers: {
              'Content-Type': 'application/json'
            },
            body: JSON.stringify({ id: task.id,title: changedTaskTitle,category: task.category}),
        }
        const res = await fetch(URL + '/' + task.id, options )
        const updatedTask = await res.json();
        editTask(updatedTask)
        changeEditStatus(false)
    } 

    return(
        <div
            key={task.id}
            onDragStart= { (event) => onDragStart(event,task.title)} 
            draggable
            className="draggable"
            >
            {changedTaskTitle}
                <div className="editSection">
                    
                    {
                        editStatus ? (
                            <>
                                <input
                                className="newInputField" 
                                placeholder={"Enter new value"}
                                onChange= {(event) => changeTaskTitle(event)}
                                />
                                <button className="saveButton" onClick={() => saveChanges()} >Save</button>
                            </>
                        ) : (
                            <div className="editButtonContainer">
                                <button onClick={() => enableEdit()}>Edit</button>
                            </div>
                            )
                    }
                </div>
            
            </div>
    )
}

export default DraggableComponent;