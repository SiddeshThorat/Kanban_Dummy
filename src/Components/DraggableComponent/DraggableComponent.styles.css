.draggable{
	width: 200px;
	background-color: rgb(163, 163, 195);
	margin: 5px auto;
	line-height: 4em;
	border-radius: 0 25px 25px 25px;
	font-size:15px;
}