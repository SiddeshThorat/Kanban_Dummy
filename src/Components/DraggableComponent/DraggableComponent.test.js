import React from 'react';
import { shallow } from 'enzyme'
import DraggableComponent from'./DraggableComponent.component';

describe("Testing Draggable component",() => {
    const props = {
        task: {id:"1",title:"Dummy Task",category: "icebox"},
        onDragStart: jest.fn(),
        editTask: jest.fn()
    }
    it("Snapshot testing of Draggable component",() => {
        expect(shallow(<DraggableComponent {...props}/>)).toMatchSnapshot();
    })
})


