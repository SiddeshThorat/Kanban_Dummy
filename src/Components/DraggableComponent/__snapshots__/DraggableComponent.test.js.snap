// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Draggable component Snapshot testing of Draggable component 1`] = `
<div
  className="draggable"
  draggable={true}
  key="1"
  onDragStart={[Function]}
>
  Dummy Task
  <div
    className="editSection"
  >
    <div
      className="editButtonContainer"
    >
      <button
        onClick={[Function]}
      >
        Edit
      </button>
    </div>
  </div>
</div>
`;
